includes/class-sensei.php

```diff
@@ -918,80 +919,158 @@ public function load_class( $class_name = '' ) {
 	} // End load_class()
 
 	/**
-	 * Filtering wp_count_comments to ensure that Sensei comments are ignored
+	 * Filtering wp_count_comments to ensure that Sensei comments are ignored.
 	 *
 	 * @since   1.4.0
 	 * @access  public
-	 * @param  array   $comments
-	 * @param  integer $post_id
-	 * @return array
+	 *
+	 * @param stdClass $comment_counts Counts by comment status.
+	 * @param int      $post_id        Post ID.
+	 *
+	 * @return stdClass
+	 */
+	public function sensei_count_comments( $comment_counts, $post_id ) {
+		if (
+			! empty( $post_id )
+			&& ! in_array( get_post_type( $post_id ), [ 'course', 'lesson', 'quiz' ], true )
+		) {
+			return $comment_counts;
+		}
+
+		$sensei_counts = $this->get_sensei_comment_counts( $post_id );
+		if ( empty( $sensei_counts ) ) {
+			return $comment_counts;
+		}
+
+		if ( empty( $comment_counts ) ) {
+			$comment_counts = $this->get_comment_count_raw();
+
+			if ( empty( $comment_counts ) ) {
+				return $comment_counts;
+			}
+		}
+
+		// Subtract Sensei's comment counts from all the comment counts.
+		foreach ( array_keys( (array) $comment_counts ) as $count_type ) {
+			if ( isset( $sensei_counts[ $count_type ] ) ) {
+				$comment_counts->{$count_type} = max( 0, (int) $comment_counts->{$count_type} - $sensei_counts[ $count_type ] );
+			}
+		}
+
+		return $comment_counts;
+	}
+
+	/**
+	 * Clear the comment count cache.
+	 *
+	 * @param int $post_id Post ID.
 	 */
-	public function sensei_count_comments( $comments, $post_id ) {
-		global $wpdb;
+	public function flush_comment_count_cache( $post_id ) {
+		$post_id = (int) $post_id;
+
+		$post_transient_id = self::COMMENT_COUNT_TRANSIENT_PREFIX . $post_id;
+		delete_transient( $post_transient_id );
 
+		$all_transient_id = self::COMMENT_COUNT_TRANSIENT_PREFIX . '0';
+		delete_transient( $all_transient_id );
+	}
+
+	/**
+	 * Get the Sensei related comment counts by comment_approved but use caching.
+	 *
+	 * @param int $post_id Post ID.
+	 *
+	 * @return array
+	 */
+	private function get_sensei_comment_counts( $post_id ) {
 		$post_id = (int) $post_id;
 
-		$count = wp_cache_get( "comments-{$post_id}", 'counts' );
+		$transient_id = self::COMMENT_COUNT_TRANSIENT_PREFIX . $post_id;
+		$stats        = get_transient( $transient_id );
 
-		if ( false !== $count ) {
-			return $count;
-		}
+		if ( ! $stats || ! is_array( $stats ) ) {
+			$stats = $this->get_sensei_comment_counts_direct( $post_id );
 
-		$statuses = array( '' ); // Default to the WP normal comments
-		// WC excludes these so exclude them too
-		$wc_statuses_to_exclude = array( 'order_note', 'webhook_delivery' );
```

Comment:
We no longer have to handle WooCommerce's junk. We hook in after WooCommerce and further remove just our own counts.